src/reducers/versions.tsx

```diff
@@ -470,46 +475,43 @@ export const getDiffAnchors = (diff: DiffInfo): string[] => {
 };
 
 export type GetRelativeDiffAnchorParams = {
-  currentAnchor: string | void;
+  currentAnchor?: string | void;
   diff: DiffInfo;
-  position: RelativePathPosition;
+  position?: RelativePathPosition;
 };
 
 export const getRelativeDiffAnchor = ({
   currentAnchor,
   diff,
-  position,
+  position = RelativePathPosition.next,
 }: GetRelativeDiffAnchorParams): string | null => {
   const anchors = getDiffAnchors(diff);
-  if (!anchors.length) {
-    // There are no changes in the diff.
-    // This can happen if a file is saved but not changed internally.
-    return null;
-  }
+  if (anchors.length) {
+    let newIndex;
+    if (!currentAnchor) {
+      // Since we aren't looking for an anchor relative to an existing one,
+      // just get the first anchor.
+      newIndex = 0;
+    } else {
+      const currentIndex = anchors.indexOf(currentAnchor);
+      if (currentIndex < 0) {
+        throw new Error(
+          `Could not locate anchor: ${currentAnchor} in the diff.`,
+        );
+      }
 
-  let newIndex;
-  if (!currentAnchor) {
-    // Since we aren't looking for an anchor relative to an existing one,
-    // just get the first anchor.
-    newIndex = 0;
-  } else {
-    const currentIndex = anchors.indexOf(currentAnchor);
-    if (currentIndex < 0) {
-      throw new Error(`Could not locate anchor: ${currentAnchor} in the diff.`);
+      newIndex =
+        position === RelativePathPosition.previous
+          ? currentIndex - 1
```

Comment:
What if `currentIndex` is equal to `0`?

Reply:
That's dealt with by the `if` on line 509 which checks `if (newIndex >= 0 && newIndex < anchors.length)`. In the case of `currentIndex` being equal to `0`, that `if` evaluates to `false` (because `newIndex` would be `-1`) and therefore we fall out the bottom of the larger `if` and return `null` at line 515.

Reply:
@willdurand I'll hold off merging this so you can take a look and see if that addresses your concern.